lib/src/utils/parsing/parsing_utils.dart

```diff
@@ -0,0 +1,321 @@
+import 'package:ledger_cardano/src/errors/invalid_data_reason.dart';
+import 'package:ledger_cardano/src/models/network.dart';
+import 'package:ledger_cardano/src/models/parsed_input.dart';
+import 'package:ledger_cardano/src/models/parsed_network.dart';
+import 'package:ledger_cardano/src/models/parsed_output.dart';
+import 'package:ledger_cardano/src/models/parsed_transaction.dart';
+import 'package:ledger_cardano/src/models/transaction.dart';
+import 'package:ledger_cardano/src/models/tx_input.dart';
+import 'package:ledger_cardano/src/models/tx_output.dart';
+import 'package:ledger_cardano/src/utils/constants.dart';
+import 'package:ledger_cardano/src/utils/utilities.dart';
+
+
+bool isArray(dynamic data) => data is List;
+
+
+
+ParsedTransaction parseTransaction(Transaction tx) {
+  final network = parseNetwork(tx.network);
+  
+  // inputs
+  validate(isArray(tx.inputs), InvalidDataReason.inputsNotArray.message);
```

Comment:
You don't need this validation because Dart is strongly typed unlike javascript (and ts by extension). If the type of tx.inputs is `List` inside the `Transaction` object, than it's definitely a List and not something else.

P.s. In typescript, the types are not enforced so even if you expect something to be a "List" for example, it could easily be a String in reality/etc